src/Microsoft.ML.StandardLearners/Standard/Simple/SimpleTrainers.cs

```diff
@@ -38,38 +39,37 @@ namespace Microsoft.ML.Runtime.Learners
     /// <summary>
     /// A trainer that trains a predictor that returns random values
     /// </summary>
-    public sealed class RandomTrainer : TrainerBase<RandomPredictor>
+
+    public sealed class RandomTrainer : TrainerEstimatorBase<BinaryPredictionTransformer<RandomPredictor>, RandomPredictor>
     {
         internal const string LoadNameValue = "RandomPredictor";
         internal const string UserNameValue = "Random Predictor";
         internal const string Summary = "A toy predictor that returns a random value.";
 
         public class Arguments
         {
-            // Some sample arguments
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Learning rate", ShortName = "lr")]
-            public Float LearningRate = (Float)1.0;
-
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Some bool arg", ShortName = "boolarg")]
-            public bool BooleanArg = false;
         }
 
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
         private static readonly TrainerInfo _info = new TrainerInfo(normalization: false, caching: false);
         public override TrainerInfo Info => _info;
 
-        public RandomTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        protected override SchemaShape.Column[] OutputColumns => throw new NotImplementedException();
+
```

Comment:
That is obviously not sufficient. You need to list the columns that you are going to output.

Reply:
Same goes for the other trainer

---
In reply to: [216508453](https://github.com/dotnet/machinelearning/pull/876#discussion_r216508453) [](ancestors = 216508453)